modules/javafx.graphics/src/main/native-glass/gtk/glass_dnd.cpp

```diff
@@ -135,9 +135,10 @@ static gboolean dnd_target_receive_data(JNIEnv *env, GdkAtom target, selection_d
 static struct {
     GdkDragContext *ctx;
     gboolean just_entered;
+    gboolean dropped;
     jobjectArray mimes;
     gint dx, dy;
-} enter_ctx = {NULL, FALSE, NULL, 0, 0};
+} enter_ctx = {NULL, FALSE, FALSE, NULL, 0, 0};
```

Comment:
it's funny how it's initialized here and also in reset_enter_ctx(), and just by coincidence both produce the same result.  This will break if initial values are other than NULL, 0, FALSE.

Reply:
Indeed. During my review, before I looked more closely, I was all ready to post a comment to the effect that it was never set back to FALSE. Then I saw the `memset` in `reset_enter_ctx` that you referred to. Not the way I would have initially designed it -- relying on '0' being the desired default value for all fields (including pointers) -- but given that it is that way, this doesn't make the problem any worse.

Reply:
I plan to gradually improve it.